integration-tests/tester/framework/network.go

```diff
@@ -0,0 +1,260 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/iotaledger/hive.go/crypto/ed25519"
+	"github.com/iotaledger/hive.go/identity"
+)
+
+type NetworkType byte
+
+const (
+	// Defines a network which consists out of autopeered nodes.
+	NetworkTypeAutopeered NetworkType = iota
+	// Defines a network which consists out of statically peered nodes.
+	NetworkTypeStatic
+)
+
+// Network is a network consisting out of Hornet nodes.
+type Network struct {
+	// The ID of the network.
+	ID string
+	// the type of the network.
+	NetworkType NetworkType
+	// The name of the network.
+	Name string
+	// The nodes within the network in the order in which they were spawned.
+	Nodes []*Node
+	// The tester docker container executing the tests.
+	tester *DockerContainer
+	// The docker client used to communicate with the docker daemon.
+	dockerClient *client.Client
+}
+
+// PrefixName returns the suffix prefixed with the name.
+func (n *Network) PrefixName(suffix string) string {
+	return fmt.Sprintf("%s-%s", n.Name, suffix)
+}
+
+// AwaitOnline awaits until all nodes are online or the given context is canceled.
+func (n *Network) AwaitOnline(ctx context.Context) error {
+	log.Println("waiting for nodes to become online...")
+	for _, node := range n.Nodes {
+		for {
+			if err := returnErrIfCtxDone(ctx, ErrNodesNotOnlineInTime); err != nil {
+				return err
+			}
+			if _, err := node.WebAPI.GetNodeInfo(); err != nil {
+				continue
+			}
+			break
+		}
+	}
+	return nil
+}
+
+// AwaitAllSync awaits until all nodes see themselves as synced.
+func (n *Network) AwaitAllSync(ctx context.Context) error {
+	log.Println("waiting for nodes to become synced...")
+	for _, node := range n.Nodes {
+		for {
+			if err := returnErrIfCtxDone(ctx, ErrNodesDidNotSyncInTime); err != nil {
```

Comment:
Small time sleep here?

Reply:
At worst one core would spin up to 100%. I won't put a sleep.